### PR TITLE
4.2.0-release-notes

### PR DESCRIPTION
**Pull request by hmottestad**
Javadocs, release-notes and news item for 4.2.0

### CODE REVIEW

File: site/content/news/rdf4j-420.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
+---
+title: "RDF4J 4.2.0 released"
+date: 2022-09-16T21:19:19+0200
+layout: "single"
+categories: ["news"]
+---
+RDF4J 4.2.0 is now available. This is a patch release fixing 8 bugs.
```
Comment (jeenbroekstra) [inline site/content/news/rdf4j-420.md:7]: It's a minor release, right, not a patch one? Might be worth highlighting the performance improvements and of course the custom aggregate feature.

File: site/content/release-notes/4.2.0.md [hunk 2]
```diff
@@ -0,0 +1,11 @@
+---
+title: "4.2.0"
+toc: true
+---
+RDF4J 4.2.0 is a patch release that fixes 8 issues.
```
Comment (jeenbroekstra) [inline site/content/release-notes/4.2.0.md:5]: Same here.

File: site/content/news/rdf4j-420.md (hunk 1 above)
Comment (hmottestad) [reply site/content/news/rdf4j-420.md:7]: It's just the autogenerated stuff at the moment. Haven't had time to sit down and write this up yet.

**Comment by hmottestad** [comment]
I've not been able to work on this until now. I have asked Tomas to add some documentation for the new custom aggregate function feature, but I propose that we publish the news item and release notes without that since I should have done that about two weeks ago already.

**Pull request merged**